sources/executor/pipe/pipe_manager.c

```diff
@@ -29,21 +29,27 @@ static void redirect_out(t_cmd *cmd)
 	}
 }
 
-void	dup_fds(t_pipes *pipes, t_cmd *cmd)
+t_cmd	*prepare_infiles(t_cmd *cmd)
```

Comment:
I had some challenges bringing this functionality out of the `dup` method. I will run some testing on this.